watch.mjs

```diff
@@ -99,6 +99,25 @@ function runWatcher(dir, name, watchTask = "tsc:watch") {
   );
 }
 
+/**
+ * Returns a new, "debounced" function that defers the execution of `fn`
+ * to the end of the event loop, such that if this function
+ * is called multiple times in a row, `fn` will only be called once.
+ * @param { () => void } fn
+ */
+function debounce(fn) {
+  let dirty = true;
+  return () => {
+    dirty = true;
+    setTimeout(() => {
+      if (dirty) {
+        dirty = false;
+        fn();
+      }
+    });
+  };
+}
+
```

Comment:
Is something like the below too naive? Seems like it would be clearer to those not steeped in the JS 'task' internals.

```js
// Minor delay to ensure all changes flush to disk before starting a build if 
// saving multiple files close together (e.g. a formatter running over a directory)
const buildDelayMs = 50;

let buildPending = 0; // timeout handle to track if a build is already queued
function onChangeNotification() {
    if (buildPending) return; // Already queued
    buildPending = setTimeout(() => {
        // The build task runs sychronously, so we can clear the timeout handle and
        // run the build knowing that nothing will interleave with those operations
        buildPending = 0;
        build();
    }, buildDelayMs);
}
```

Reply:
I blended our two changes. Your variable names and comments are better and the timeout is a good idea. And I guess my "function that takes a function" approach wasn't popular so I got rid of it ;) 

I stuck with using a `bool` instead of the "handle" in your code - seemed more straightforward to me, and also I get a type error with your code since `setTimeout` doesn't return a number.

I also increased the delay to 100ms. In VS Code, every save action results in two file changes due to the "format on save" feature, and 50ms wasn't long enough to batch those together for me. 100ms did the trick.

Reply:
The `if (buildPending)` check is probably redundant, but doesn't hurt to be cautious. Thanks!